src/main/java/com/google/idea/perf/Matching.kt

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.idea.perf
+
+class MatchResult(
+    val source: String,
+    val matchedChars: List<Int>,
+    val score: Int
+)
+
+fun fuzzyMatchMany(sources: Collection<String>, pattern: String): List<MatchResult> {
+    val matches = ArrayList<MatchResult>(sources.size)
+
+    for (source in sources) {
+        matches.add(fuzzyMatch(source, pattern))
+    }
+
+    matches.sortByDescending { it.score }
+    return matches
```

Comment:
Optional: this method could be more concise as
```kotlin
return sources
    .map { fuzzyMatch(it, pattern) }
    .sortedByDescending { it.score }
```
(Unless you're trying to reduce memory allocations?)